conjure-java-core/src/integrationInput/java/com/palantir/product/ManyFieldExample.java

```diff
@@ -298,7 +298,8 @@ public Builder optionalItem(@Nonnull String optionalItem) {
         @JsonSetter(value = "items", nulls = Nulls.SKIP, contentNulls = Nulls.FAIL)
         public Builder items(@Nonnull Iterable<String> items) {
             checkNotBuilt();
-            this.items = ConjureCollections.newArrayList(Preconditions.checkNotNull(items, "items cannot be null"));
+            this.items =
+                    ConjureCollections.newNonNullArrayList(Preconditions.checkNotNull(items, "items cannot be null"));
```

Comment:
Let's add a unit test which asserts a NPE is thrown when we execute the following:

```java
CovariantListExample.builder().items(Collections.singletonList(null));
CovariantListExample.builder().addAllItems(Collections.singletonList(null));
CovariantListExample.builder().items((String) null);
```